ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -138,17 +153,22 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
             connId = selector.connect(new InetSocketAddress(host, port.getPort()), networkConfig.socketSendBufferBytes,
                 networkConfig.socketReceiveBufferBytes, port.getPortType());
             connectionTracker.startTrackingInitiatedConnection(host, port, connId);
+            requestMetadata.connectionCheckOutAttempts++;
           }
         } else {
-          sends.add(new NetworkSend(connId, requestMetadata.requestInfo.getRequest(), null, time));
+          sends.add(new NetworkSend(connId, requestMetadata.requestInfo.getRequest(),
+              requestMetadata.clientNetworkRequestMetrics, time));
           connectionIdToRequestInFlight.put(connId, requestMetadata);
           iter.remove();
+          numPendingConnections.set(pendingRequests.size());
```

Comment:
aah... good catch. Let me think about it. Do you have any inclination towards one of them ?